hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/PNCounterGeneratorCodecTemplate.java

```diff
@@ -17,36 +17,72 @@
 
     /**
      * Query operation to retrieve the current value of the PNCounter.
+     * <p>
+     * The invocation will return the replica timestamps (vector clock) which
+     * can then be sent with the next invocation to keep session consistency
+     * guarantees.
+     * The target replica is determined by the {@code targetReplica} parameter.
+     * If smart routing is disabled, the actual member processing the client
+     * message may act as a proxy.
      *
-     * @param name the name of the PNCounter
+     * @param name              the name of the PNCounter
+     * @param replicaTimestamps last observed replica timestamps (vector clock)
+     * @param targetReplica     the target replica
      * @return the current value of the counter
      */
-    @Request(id = 1, retryable = true, response = ResponseMessageConst.LONG)
+    @Request(id = 1, retryable = true, response = ResponseMessageConst.CRDT_TIMESTAMPED_LONG)
     @Since("1.6")
-    Object get(String name);
+    Object get(String name, @Nullable List<Entry<String, Long>> replicaTimestamps, Address targetReplica);
 
     /**
      * Adds a delta to the PNCounter value. The delta may be negative for a
      * subtraction.
+     * <p>
+     * The invocation will return the replica timestamps (vector clock) which
+     * can then be sent with the next invocation to keep session consistency
+     * guarantees.
+     * The target replica is determined by the {@code targetReplica} parameter.
+     * If smart routing is disabled, the actual member processing the client
+     * message may act as a proxy.
      *
-     * @param name            the name of the PNCounter
-     * @param delta           the delta to add to the counter value, can be negative
-     * @param getBeforeUpdate {@code true} if the operation should return the
-     *                        counter value before the addition, {@code false}
-     *                        if it should return the value after the addition
+     * @param name              the name of the PNCounter
+     * @param delta             the delta to add to the counter value, can be negative
+     * @param getBeforeUpdate   {@code true} if the operation should return the
+     *                          counter value before the addition, {@code false}
+     *                          if it should return the value after the addition
+     * @param replicaTimestamps last observed replica timestamps (vector clock)
+     * @param targetReplica     the target replica
      * @return the value before the update if {@code getBeforeUpdate} is
      * {@code true}, otherwise the value after the update
      */
-    @Request(id = 2, retryable = false, response = ResponseMessageConst.LONG)
+    @Request(id = 2, retryable = false, response = ResponseMessageConst.CRDT_TIMESTAMPED_LONG)
+    @Since("1.6")
+    Object add(String name, long delta, boolean getBeforeUpdate, @Nullable List<Entry<String, Long>> replicaTimestamps, Address targetReplica);
```

Comment:
@mmedenjak 
Similar comments for this Nullable also
see https://github.com/hazelcast/hazelcast-client-protocol/pull/113/files#r165902493

Reply:
There's no difference, used `null` to transfer as little data as possible.

Reply:
Updated.